packages/flutter/lib/src/sentry_privacy_options.dart

```diff
@@ -162,6 +170,56 @@ class SentryPrivacyOptions {
   }
 }
 
+/// Returns `true` if a SensitiveContent masking rule _should_ be added for a
+/// given [flutterVersion] string. The SensitiveContent widget was introduced
+/// in Flutter 3.33, therefore we only add the masking rule when the detected
+/// version is >= 3.33.
+bool _shouldAddSensitiveContentRule(String version) {
+  final dot = version.indexOf('.');
+  if (dot == -1) return false;
+
+  final major = int.tryParse(version.substring(0, dot));
+  final nextDot = version.indexOf('.', dot + 1);
+  final minor = int.tryParse(
+      version.substring(dot + 1, nextDot == -1 ? version.length : nextDot));
+
+  return major != null &&
```

Comment:
nit: Could do early returns when either major/minor is null. Also, do we do version check anywhere else? Then we could extract the version reading to a common place and only do the version check here, as we do both now.